starter-code/public/scripts/articleView.js

```diff
@@ -2,142 +2,151 @@
-  articleView.handleAuthorFilter();
-  articleView.handleMainNav();
-  articleView.setTeasers();
-  $('pre code').each((i, block) => hljs.highlightBlock(block));
-};
-
-articleView.initAdminPage = function() {
-  // TODO: Call the Handlebars `.compile` function, which will return a function for you to use where needed.
-  // Make sure you assign the result of your Handlebars.compile call to a variable called "template", since
-  // we are then calling "template" on line 117.
-
-  // REVIEW: We use `forEach` here because we are relying on the side-effects of the callback function:
-  // appending to the DOM.
-  // The callback is not required to return anything.
-  Article.numWordsByAuthor().forEach(stat => $('.author-stats').append(template(stat)));
-
-  // REVIEW: Simply write the correct values to the page:
-  $('#blog-stats .articles').text(Article.all.length);
-  $('#blog-stats .words').text(Article.numWordsAll());
-};
+    });
+  };
+
+  articleView.initNewArticlePage = function() {
+    $('.tab-content').show();
+    $('#export-field').hide();
+    $('#article-json').on('focus', function(){
+      this.select();
+    });
+
+    $('#new-form').on('change', 'input, textarea', articleView.create);
+    $('#new-form').on('submit', articleView.submit);
+  };
+
+  articleView.create = function() {
+    var article;
+    $('#articles').empty();
+
+    article = new Article({
+      title: $('#article-title').val(),
+      author: $('#article-author').val(),
+      authorUrl: $('#article-author-url').val(),
+      category: $('#article-category').val(),
+      body: $('#article-body').val(),
+      publishedOn: new Date().toISOString()
+    });
+
+    $('#articles').append(article.toHtml());
+    $('pre code').each((i, block) => hljs.highlightBlock(block));
+  };
+
+  articleView.submit = function(event) {
+    event.preventDefault();
+    let article = new Article({
+      title: $('#article-title').val(),
+      author: $('#article-author').val(),
+      authorUrl: $('#article-author-url').val(),
+      category: $('#article-category').val(),
+      body: $('#article-body').val(),
+      publishedOn: new Date().toISOString()
+    });
+
+    article.insertRecord();
+  }
+
+  articleView.initIndexPage = function() {
+    Article.all.forEach(a => {
+      $('#articles').append(a.toHtml())
+    });
+
+    articleView.populateFilters();
+    articleView.handleCategoryFilter();
+    articleView.handleAuthorFilter();
+    articleView.handleMainNav();
+    articleView.setTeasers();
+    $('pre code').each((i, block) => hljs.highlightBlock(block));
+  };
+  
+  articleView.initAdminPage = function() {
+    // TODO: Call the Handlebars `.compile` function, which will return a function for you to use where needed.
+    // Make sure you assign the result of your Handlebars.compile call to a variable called "template", since
+    // we are then calling "template" on line 117.
+    var source   = $('#author-stats-template').text();
```

Comment:
Note this can be done on one line, using only one variable declaration. The app is small now, so memory is whatever, but it's good to be conscious about that sort of aspect.